### DISCUSSION TITLE
Redis-server will not start on new master node

### DISCUSSION THREAD

**Discussion by byronsims** [Security-Onion-Solutions/securityonion#3436] [Unsupported Versions]
Just got a new master node installed.  Redis-server will not start.  I'm not really sure where to start on the troubleshooting.  It gives the following error:

1:M 15 Mar 2021 02:11:31.621 # Failed to configure TLS. Check logs for more info.
1:C 15 Mar 2021 02:14:44.046 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 15 Mar 2021 02:14:44.046 # Redis version=6.0.9, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 15 Mar 2021 02:14:44.046 # Configuration loaded
1:M 15 Mar 2021 02:14:44.046 # Failed to load certificate: /certs/redis.crt: error:0200100D:system library:fopen:Permission denied

**Comment by weslambert** [comment]
What is the output of the following?

`ls -la /etc/pki`

**Comment by byronsims** [comment]
drwxr-x---   4 root     root          4096 Mar 15 01:57 .
drwxr-xr-x 115 root     root          4096 Mar 16 13:04 ..
drwxr-xr-x   2 root     root          4096 Mar 15 01:57 ca.cer
-rw-r-----   1 root     root          2078 Mar 15 01:16 ca.crt
-rw-r-----   1 root     socore        3243 Mar 15 01:16 ca.key
-rw-r-----   1 root     root          2102 Mar 15 01:17 elasticsearch.crt
-rw-r-----   1 root     elasticsearch 3243 Mar 15 01:17 elasticsearch.key
-rw-r-----   1 root     elasticsearch 4269 Mar 15 01:17 elasticsearch.p12
-rw-r-----   1 root     root          2069 Mar 15 01:17 filebeat.crt
-rw-r-----   1 root     socore        3243 Mar 15 01:17 filebeat.key
-rw-r-----   1 logstash socore        3272 Mar 15 01:17 filebeat.p8
-rw-r-----   1 root     root          1720 Mar 15 01:17 fleet.crt
-rw-r-----   1 root     socore        3247 Mar 15 01:17 fleet.key
-rw-r-----   1 root     root          2102 Mar 15 01:17 influxdb.crt
-rw-r-----   1 root     socore        3243 Mar 15 01:17 influxdb.key
drwxr-x---   2 root     root          4096 Mar 15 01:17 issued_certs
-rw-r-----   1 root     root          2143 Mar 15 01:17 managerssl.crt
-rw-r-----   1 root     socore        3247 Mar 15 01:17 managerssl.key
-rw-r-----   1 root     root          2102 Mar 15 01:17 minio.crt
-rw-r-----   1 root     socore        3243 Mar 15 01:17 minio.key
-rw-r-----   1 root     root          2102 Mar 15 01:17 redis.crt
-rw-r-----   1 root     socore        3247 Mar 15 01:17 redis.key
-rw-r-----   1 root     root          2102 Mar 15 01:17 registry.crt
-rw-r-----   1 root     socore        3243 Mar 15 01:17 registry.key

**Reply by weslambert** [reply]
Is this box STIG'd, or has the default umask or have other permissions been modified?

**Comment by byronsims** [comment]
It will have the CSI benchmark configs done to it.

**Reply by weslambert** [reply]
There are probably going to be some issues with regard to umask and other settings with this approach.  Unfortunately, we don't support fully STIG'd or CIS'd boxes at the moment.  This is something that we are planning to look into, as we realize there is certainly a need.  I would recommend you first start with a non-benchmarked box, then apply changes.  Again, this wouldn't really be supported, but will help you to hone in on the various complications associated with the various CIS benchmark settings.  To ensure everything works as you expect, I would use a stock Ubuntu 18.04 or Centos 7 install, or install from the Security Onion ISO image.
